tools/test_modules/m25400.pm

```diff
@@ -171,32 +171,41 @@ sub module_generate_hash
 
 
   ################ USER PASSWORD #################
-  my $res = pdf_compute_encryption_key_user($word, $padding, $id, $u, $o, $P, $V, $R, $enc);
-
-  my $digest = md5 ($padding . pack ("H*", $id));
+  # do not change $u if it exists, keep this the same, as we don't know the user password,
+  #  we cannot calculate this part of the hash again
+  if("".$u eq "0000000000000000000000000000000000000000000000000000000000000000")
```

Comment:
why do you use this notation ?
I think
```
if ($u eq "0000000000000000000000000000000000000000000000000000000000000000")
```
would be much better

(also please use our code style with spaces after the "if". thx

Reply:
I agree, fixed both